### PR TITLE
Add Vertex input attrib check and layout Type Check

### PR DESCRIPTION
**Pull request by ShchchowAMD**
Add related constant:
1. gl_MaxVaryingVectors

Add following checks:
1. vertex varying in input attribs check
2. symbol location check should not be null
3.  Comma op should be folded as aggregate,
otherwise, constant initialization would fail
see new case at the end of 100.frag

### CODE REVIEW

**Comment by johnkslang** [comment]
This needs a bit more explanation.

The specification says:

> However, the lowest precedence operators of the sequence operator (,) and the
assignment operators (=, +=, …) are not included in the operators that can create a constant
expression.

Are you are trying to undo this rule?

**Comment by ShchchowAMD** [comment]
Hi John,

I re-check the spec and I think you are right.

From es spec,
Constant expression excludes comma from version > 100 (as sequence operator).

This rule was introduced with version > 100,

So I add version check and also test cases now.

**Comment by johnkslang** [comment]
No, it was fixed in the spec because the spec was wrong, not because version 100 should actually support the comma.

**Comment by ShchchowAMD** [comment]
Thanks for the explanation.
I've removed comma part and test in version 100.

Now we will treat this rule as applying under all versions.

**Comment by ShchchowAMD** [comment]
Hi John, is there any further suggestions for this change please?

File: glslang/MachineIndependent/Intermediate.cpp [hunk 1]
```diff
@@ -2450,7 +2450,7 @@ TIntermTyped* TIntermediate::addComma(TIntermTyped* left, TIntermTyped* right, c
     commaAggregate->setType(right->getType());
     commaAggregate->getWritableType().getQualifier().makeTemporary();
 
-    return commaAggregate;
+    return commaAggregate->getAsAggregate();
```
Comment (johnkslang) [inline glslang/MachineIndependent/Intermediate.cpp:2453]: Is this change still needed?  It seems to add something redundant.

File: glslang/MachineIndependent/Constant.cpp [hunk 2]
```diff
@@ -1047,6 +1047,8 @@ TIntermTyped* TIntermediate::fold(TIntermAggregate* aggrNode)
         objectSize = std::max(children[0]->getAsTyped()->getType().getVectorSize(),
                               children[2]->getAsTyped()->getType().getVectorSize());
         break;
+    case EOpComma:
+        return children.back()->getAsTyped();
```
Comment (johnkslang) [inline glslang/MachineIndependent/Constant.cpp:1051]: Should this still be included?

**Comment by ShchchowAMD** [comment]
Close current version, comma case needs to be removed with further support.
Will be moved to another new pr as this has been placed in PR list too long.
